workflow/controller/operator.go

```diff
@@ -1037,6 +1037,36 @@ func (woc *wfOperationCtx) assessNodeStatus(pod *apiv1.Pod, node *wfv1.NodeStatu
 			node.Phase = wfv1.NodeError
 		} else {
 			node.Outputs = &outputs
+			tmplCtx, err := woc.createTemplateContext(node.GetTemplateScope())
+			if err != nil {
+				log.Errorf("Failed to create template context for node %s", node.ID)
+				return nil
+			}
+			_, resolvedTmpl, _, err := tmplCtx.ResolveTemplate(node)
+			if err != nil {
+				log.Errorf("Failed to resolve template for node %s: %s", node.ID, err)
+				return nil
+			}
+
+			localParams := make(map[string]string)
+			_, localArgs, _ := woc.loadExecutionSpec()
+			woc.setGlobalParameters(localArgs)
+			// Inject the pod name. If the pod has a retry strategy, the pod name will be changed and will be injected when it
+			// is determined
+			if resolvedTmpl.IsPodType() && resolvedTmpl.RetryStrategy == nil {
+				localParams[common.LocalVarPodName] = woc.wf.NodeID(node.Name)
+			}
+
+			tmpl, err := common.ProcessArgs(resolvedTmpl, &localArgs, woc.globalParams, localParams, false)
+			if err != nil {
+				log.Errorf("Failed to process template for node %s: %s", node.ID, err)
+			}
+
+			if tmpl.Memoize != nil {
+				c := woc.controller.cache.(*configMapCache)
+				c.configMapName = tmpl.Memoize.Cache.ConfigMapName.Name
```

Comment:
Struct variables should not be set like this. I think the`configMapName` should actually be managed by the controller and not set by the user. Let's discuss this.